Solaris11 general build errors

I getting multiple errors while building 
*Upstream LibreSSL (github master)

while executing ./autogen.sh (awk error, see logfile)
*Uninstalling/ReInstalling awk or greplacing it with Gawk doesnt help on Solaris11.4

Using:
https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.8.3.tar.gz
and 
https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.9.0.tar.gz

Results in build error:
  CC       libcrypto_la-cpt_err.lo
In file included from /usr/include/sys/time.h:447:0,
                 from ../include/compat/sys/time.h:13,
                 from /usr/include/sys/select.h:27,
                 from ../include/compat/sys/select.h:7,
                 from /usr/include/sys/types.h:665,
                 from ../include/compat/sys/types.h:13,
                 from /usr/include/sys/wait.h:12,
                 from /usr/include/stdlib.h:16,
                 from ../include/compat/stdlib.h:13,
                 from ../include/openssl/err.h:118,
                 from cpt_err.c:65:
/usr/include/sys/timespec_util.h:29:13: error: expected identifier or '(' before 'do'
 extern void timespecsub(const struct timespec *t1, const struct timespec *t2,
             ^
/usr/include/sys/timespec_util.h:29:13: error: expected identifier or '(' before 'while'
 extern void timespecsub(const struct timespec *t1, const struct timespec *t2,
[solaris11_autogen.sh_upstream_github_awk_step_error.log](https://github.com/libressl-portable/portable/files/2729590/solaris11_autogen.sh_upstream_github_awk_step_error.log)
[solaris11_gnu_make_after_cmake_compile_error-libressl-2.9.0.tar.gz.log](https://github.com/libressl-portable/portable/files/2729591/solaris11_gnu_make_after_cmake_compile_error-libressl-2.9.0.tar.gz.log)
[solaris11_gnu_make_compile_error-libressl-2.9.0.tar.gz.log](https://github.com/libressl-portable/portable/files/2729592/solaris11_gnu_make_compile_error-libressl-2.9.0.tar.gz.log)
[solaris11_gnu_make_compile_error-libressl-2.8.3.tar.gz.log](https://github.com/libressl-portable/portable/files/2729593/solaris11_gnu_make_compile_error-libressl-2.8.3.tar.gz.log)
[solaris11_gnu_make_after_cmake_compile_error-libressl-2.8.3.tar.gz.log](https://github.com/libressl-portable/portable/files/2729594/solaris11_gnu_make_after_cmake_compile_error-libressl-2.8.3.tar.gz.log)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -237,6 +237,11 @@ if(HAVE_GETPAGESIZE)
 	add_definitions(-DHAVE_GETPAGESIZE)
 endif()
 
+check_symbol_exists(timespecsub sys/time.h HAVE_TIMESPECSUB)
+if(HAVE_TIMESPECSUB)
+	add_definitions(-DHAVE_TIMESPECSUB)
+endif()
+
 check_function_exists(timingsafe_bcmp HAVE_TIMINGSAFE_BCMP)
 if(HAVE_TIMINGSAFE_BCMP)
 	add_definitions(-DHAVE_TIMINGSAFE_BCMP)

--- a/include/compat/time.h
+++ b/include/compat/time.h
@@ -39,7 +39,11 @@ typedef int clockid_t;
 int clock_gettime(clockid_t clock_id, struct timespec *tp);
 #endif
 
-#ifndef timespecsub
+#ifdef timespecsub
+#define HAVE_TIMESPECSUB
+#endif
+
+#ifndef HAVE_TIMESPECSUB
 #define timespecsub(tsp, usp, vsp)                                      \
         do {                                                            \
                 (vsp)->tv_sec = (tsp)->tv_sec - (usp)->tv_sec;          \
@@ -50,6 +54,7 @@ int clock_gettime(clockid_t clock_id, struct timespec *tp);
                 }                                                       \
         } while (0)
 #endif
+
 #endif
 
 #endif

--- a/m4/check-libc.m4
+++ b/m4/check-libc.m4
@@ -22,7 +22,6 @@ AC_CACHE_CHECK([for getpagesize], ac_cv_func_getpagesize, [
 AM_CONDITIONAL([HAVE_ASPRINTF], [test "x$ac_cv_func_asprintf" = xyes])
 AM_CONDITIONAL([HAVE_FREEZERO], [test "x$ac_cv_func_freezero" = xyes])
 AM_CONDITIONAL([HAVE_GETPAGESIZE], [test "x$ac_cv_func_getpagesize" = xyes])
-AM_CONDITIONAL([HAVE_TIMESPECSUB], [test "x$ac_cv_func_timespecsub" = xyes])
 AM_CONDITIONAL([HAVE_MEMMEM], [test "x$ac_cv_func_memmem" = xyes])
 AM_CONDITIONAL([HAVE_READPASSPHRASE], [test "x$ac_cv_func_readpassphrase" = xyes])
 AM_CONDITIONAL([HAVE_REALLOCARRAY], [test "x$ac_cv_func_reallocarray" = xyes])

--- a/m4/check-libc.m4
+++ b/m4/check-libc.m4
@@ -5,7 +5,7 @@ AC_CHECK_HEADERS([err.h readpassphrase.h])
 AC_CHECK_FUNCS([asprintf freezero memmem])
 AC_CHECK_FUNCS([readpassphrase reallocarray recallocarray])
 AC_CHECK_FUNCS([strlcat strlcpy strndup strnlen strsep strtonum])
-AC_CHECK_FUNCS([timegm _mkgmtime])
+AC_CHECK_FUNCS([timegm _mkgmtime timespecsub])
 AC_CACHE_CHECK([for getpagesize], ac_cv_func_getpagesize, [
 	AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 // Since Android NDK v16 getpagesize is defined as inline inside unistd.h
@@ -22,6 +22,7 @@ AC_CACHE_CHECK([for getpagesize], ac_cv_func_getpagesize, [
 AM_CONDITIONAL([HAVE_ASPRINTF], [test "x$ac_cv_func_asprintf" = xyes])
 AM_CONDITIONAL([HAVE_FREEZERO], [test "x$ac_cv_func_freezero" = xyes])
 AM_CONDITIONAL([HAVE_GETPAGESIZE], [test "x$ac_cv_func_getpagesize" = xyes])
+AM_CONDITIONAL([HAVE_TIMESPECSUB], [test "x$ac_cv_func_timespecsub" = xyes])
 AM_CONDITIONAL([HAVE_MEMMEM], [test "x$ac_cv_func_memmem" = xyes])
 AM_CONDITIONAL([HAVE_READPASSPHRASE], [test "x$ac_cv_func_readpassphrase" = xyes])
 AM_CONDITIONAL([HAVE_REALLOCARRAY], [test "x$ac_cv_func_reallocarray" = xyes])

